src/main/java/hexgui/gui/MessageDialogs.java

```diff
@@ -0,0 +1,390 @@
+        options,
+        defaultOption,
+        -1);
+  }
+
+  public void showError(Component frame, String message, Exception e) {
+    showError(frame, message, e, true);
+  }
+
+  public void showError(Component frame, String message, Exception e, boolean isCritical) {
+    showError(frame, message, StringUtils.getErrorMessage(e), isCritical);
+  }
+
+  public void showInfo(
+      Component frame, String mainMessage, String optionalMessage, boolean isCritical) {
+    showInfo(null, frame, mainMessage, optionalMessage, isCritical);
+  }
+
+  public void showInfo(
+      String disableKey,
+      Component frame,
+      String mainMessage,
+      String optionalMessage,
+      boolean isCritical) {
+    if (checkDisabled(disableKey)) return;
+    int type;
+    if (isCritical) type = JOptionPane.INFORMATION_MESSAGE;
+    else type = JOptionPane.PLAIN_MESSAGE;
+    Object[] options = {"Close"};
+    Object defaultOption = options[0];
+    String title = "Information" + " - " + m_applicationName;
+    show(
+        disableKey,
+        frame,
+        title,
+        mainMessage,
+        optionalMessage,
+        type,
+        JOptionPane.DEFAULT_OPTION,
+        options,
+        defaultOption,
+        -1);
+  }
+
+  public int showYesNoCancelQuestion(
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String destructiveOption,
+      String nonDestructiveOption) {
+    return showYesNoCancelQuestion(
+        null, parent, mainMessage, optionalMessage, destructiveOption, nonDestructiveOption);
+  }
+
+  /**
+   * Show a question with two options and cancel.
+   *
+   * @return 0 for the destructive option; 1 for the non-destructive option; 2 for cancel
+   */
+  public int showYesNoCancelQuestion(
+      String disableKey,
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String destructiveOption,
+      String nonDestructiveOption) {
+    if (checkDisabled(disableKey)) return 0;
+    Object[] options = new Object[3];
+    int destructiveIndex;
+    if (Platform.isMac()) {
+      options[0] = nonDestructiveOption;
+      options[1] = "Cancel";
+      options[2] = destructiveOption;
+      destructiveIndex = 2;
+    } else {
+      options[0] = nonDestructiveOption;
+      options[1] = destructiveOption;
+      options[2] = "Cancel";
+      destructiveIndex = -1;
+    }
+    Object defaultOption = options[0];
+    int type = JOptionPane.QUESTION_MESSAGE;
+    String title = "Question" + " - " + m_applicationName;
+    Object value =
+        show(
+            disableKey,
+            parent,
+            title,
+            mainMessage,
+            optionalMessage,
+            type,
+            JOptionPane.YES_NO_CANCEL_OPTION,
+            options,
+            defaultOption,
+            destructiveIndex);
+    int result;
+    if (value == destructiveOption) result = 0;
+    else if (value == nonDestructiveOption) result = 1;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.